src/iface-opc-ua.c

```diff
@@ -240,21 +293,22 @@ int iface_opc_ua_read_data(struct knot_data_item *data_item)
 	rc = setup_read_data(value, data_item);
 	memset(&tmp, 0, sizeof(tmp));
 
-	if (rc == UA_STATUSCODE_GOOD) {
-		rc = get_read_value(value, data_item->schema.value_type,
+	if (!((rc & UA_STATUSCODE_BAD) || (rc & UA_STATUSCODE_UNCERTAIN))) {
+		get_read_value(value, data_item->schema.value_type,
 				    data_item->value_type_size, &tmp);
 	}
 
-	if (rc & UA_STATUSCODE_BAD) {
+	if (rc & UA_STATUSCODE_BADSERVERNOTCONNECTED) {
 		l_error("Failed to read from OPC UA: %#010x",
 			rc);
 		l_io_destroy(opc_ua_io);
 		opc_ua_io = NULL;
 		rental = -EINTR;
-	} else if (rc & UA_STATUSCODE_UNCERTAIN) {
+	} else if (((rc & UA_STATUSCODE_BAD) ||
+		   (rc & UA_STATUSCODE_UNCERTAIN))) {
 		l_error("Failed to read from OPC UA for Data_%d: %#010x",
 			data_item->sensor_id, rc);
```

Comment:
I would recommend replacing the 'rc' and 'fd' variables with names with more details